openedx/core/lib/request_utils.py

```diff
@@ -102,19 +102,71 @@ def process_request(self, request):
 
         Don't log contents of cookies because that might cause a security issue.
         We just want to see if any cookies are growing out of control.
+
+        A useful NRQL Query:
+            SELECT count(*), max(`cookies.max.group.size`) from Transaction FACET
+            `cookies.max.group.name`
+
+            SELECT * FROM Transaction WHERE cookies_total_size > 6000
+
+        Attributes that are added by this middleware:
+
+        cookies.<cookie_name>.size: The size of a cookie by the given name.
+        cookies.<group_prefix>.group.size: The size of a group of cookies. For example
+            the sum of the size of all braze cookies would be the value of the
+            `cookies.ab.group.size` attribute.
+        cookies.max.name: The name of the largest cookie sent by the user.
+        cookies.max.size: The size of the largest cookie sent by the user.
+        cookies.max.group.name: The name of the largest group of cookies. A single cookie
+            counts as a group of one for this calculation.
+        cookies.max.group.size: The sum total size of all the cookies in the largest group.
+        cookies_total_size: The sum total size of all cookies in this request.
+
         """
         if not CAPTURE_COOKIE_SIZES.is_enabled():
             return
 
-        cookie_names_to_size = {
-            name: len(value)
-            for name, value in request.COOKIES.items()
-        }
+        cookie_names_to_size = {}
+        cookie_groups_to_size = {}
+
+        for name, value in request.COOKIES.items():
+            # Get cookie size for all cookies.
+            cookie_size = len(value)
+            cookie_names_to_size[name] = cookie_size
+
+            # Group cookies by their prefix seperated by a period or underscore
+            grouping_name = re.split('[._]', name, 1)[0]
+            if grouping_name and grouping_name != name:
+                # Add or update the size for this group.
+                cookie_groups_to_size[grouping_name] = cookie_groups_to_size.get(grouping_name, 0) + cookie_size
+
+        max_cookie_name = max(cookie_names_to_size, key=lambda name: cookie_names_to_size[name])
+        max_cookie_size = cookie_names_to_size[max_cookie_name]
+
+        max_group_cookie_name = max(cookie_groups_to_size, key=lambda name: cookie_groups_to_size[name])
+        max_group_cookie_size = cookie_groups_to_size[max_group_cookie_name]
+
+        # If a single cookies is bigger than any group of cookies, we want max_group... to reflect that.
+        # Treating an individual cookie as a group of 1 for calculating the max.
+        if max_group_cookie_size < max_cookie_size:
+            max_group_cookie_name = max_cookie_name
+            max_group_cookie_size = max_cookie_size
+
         for name, size in cookie_names_to_size.items():
             attribute_name = 'cookies.{}.size'.format(name)
```

Comment:
Let's stop logging individual cookies by name if it's going to be hashed on a per-user basis like some of our cookie providers were doing. It's going to explode the metrics listing out too much. The consolidated by itself should be enough.

Reply:
Maybe in another PR? Otherwise, we don’t have the full cookie names related to the group. Want to get through CAT-1 first before improving this.

Reply:
@feanil: Based on the issue I found with New Relic dropping data, I agree that we need to make a change.  However, I think if we drop all the individual cookies, we will lose too much potentially useful data about what to improve next. I propose we skip any individual cookie and any group total that doesn't hit some threshold.  We could hard-code to 100 or 150 to start and/or make the value a Django Setting.  What do you think?

Here is a query to see some examples of the data we were seeing: https://one.nr/0mMRNEEzAwn